webpack/assets/javascripts/react_app/components/ForemanModal/ConnectedForemanModal.test.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Provider } from 'react-redux';
+import { Modal } from 'patternfly-react';
+import { mount } from 'enzyme';
+import configureMockStore from 'redux-mock-store';
+import ForemanModal from './index';
+import ForemanModalHeader from './subcomponents/ForemanModalHeader';
+import ForemanModalFooter from './subcomponents/ForemanModalFooter';
+
+// This file is for integration tests of the Redux-connected ForemanModal component (no unit tests)
+
+const middlewares = [];
+const mockStore = configureMockStore(middlewares);
+const state = {
+  foremanModals: {
+    render: { open: true },
+    closed: { open: false },
+    withSuppliedChildren: { open: true },
+    unordered: { open: true },
+    defaultHeaderChild: { open: true },
+    customHeaderChild: { open: true },
+    noHeader: { open: true },
+  },
+};
+
+const ConnectedForemanModal = props => (
+  <Provider store={mockStore(state)}>
+    <ForemanModal {...props} />
+  </Provider>
+);
+
+const headerChild = (
+  <ForemanModal.Header>this is the header</ForemanModal.Header>
+);
+const footerChild = (
+  <ForemanModal.Footer>this is the footer</ForemanModal.Footer>
+);
+const modalBody = <div>This is the body</div>;
+
+describe('ForemanModal with Redux & hooks', () => {
+  describe('rendering', () => {
```

Comment:
It doesn't feel like an integration test.

Instead of testing your component works together with your reducers, actions selectors. It render with modal with a given state like you already do in your unit testing.

IMO it should use the `IntegrationTestHelper` to make sure all the different parts work together as a bigger unit.

Just sketching around, but I expected something like:
https://gist.github.com/sharvit/b7d193c033d9c5fe6371eb4efb27a467

Reply:
yeah it's more like a fat unit test, right?  Thanks for the idea, I will play around with this..

Reply:
@sharvit Updated the tests as per your suggestion, thanks :)